pkg/controlsvc/controlsvc_test.go

```diff
@@ -0,0 +1,647 @@
+package controlsvc_test
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"io"
+	"net"
+	"os"
+	"testing"
+
+	"github.com/ansible/receptor/pkg/controlsvc"
+	"github.com/ansible/receptor/pkg/controlsvc/mock_controlsvc"
+	"github.com/ansible/receptor/pkg/logger"
+	"github.com/golang/mock/gomock"
+)
+
+const (
+	writeToConnError = "write to conn write message err"
+)
+
+func printExpectedError(t *testing.T, err error) {
+	t.Errorf("expected error %s", err)
+}
+
+func TestConnectionListener(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockNetceptor := mock_controlsvc.NewMockNetceptorForControlsvc(ctrl)
+	mockListener := mock_controlsvc.NewMockListener(ctrl)
+	logger := logger.NewReceptorLogger("")
+
+	connectionListenerTestCases := []struct {
+		name          string
+		expectedError bool
+		expectedCalls func(context.CancelFunc)
+	}{
+		{
+			name:          "return from context error",
+			expectedError: true,
+		},
+		{
+			name:          "error accepting connection",
+			expectedError: false,
+			expectedCalls: func(ctxCancel context.CancelFunc) {
+				mockListener.EXPECT().Accept().DoAndReturn(func() (net.Conn, error) {
+					ctxCancel()
+					return nil, errors.New("terminated")
+				})
+				mockNetceptor.EXPECT().GetLogger().Return(logger)
+			},
+		},
+	}
+
+	for _, testCase := range connectionListenerTestCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			ctx, ctxCancel := context.WithCancel(context.Background())
+			defer ctxCancel()
+
+			if testCase.expectedCalls != nil {
```

Comment:
I personally prefer a redundant `expectedCalls()` rather than an `if` condition. Consider using `expectedCalls: func(ctxCancel context.CancelFunc) {}` in the first test case

Reply:
if you fill in this blank `expectedCalls` then you can get rid of the `expectedError` field

Reply:
I had it the first way before but code smells wanted a comment on each empty function call. 

I can change it back with adding a meaning less comment to pass code smells.